src/utils.cpp

```diff
@@ -12,11 +12,28 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "utils.hpp"
 #include <cstdarg>
 #include <cstdio>
 #include <cstdlib>
 
-#include "utils.hpp"
+#ifdef WIN32
+#include <Windows.h>
+#include <io.h>
+char* mkdtemp_win32(char *tmpl)
+{
+    const size_t tmpl_len = std::strlen(tmpl);
```

Comment:
I guess another way of doing this would be to make `cvk_mkdtemp` a function with the #ifdef'ry inside it and pass the length of the string from `cvk_program::compile_source`. You'd avoid calling `strlen`.